src/mds/MDSAuthCaps.cc

```diff
@@ -145,7 +145,7 @@ bool MDSCapMatch::match(string_view target_path,
       bool gid_matched = false;
       if (std::find(gids.begin(), gids.end(), caller_gid) != gids.end())
 	gid_matched = true;
-      if (caller_gid_list) {
+      if (!gid_matched && caller_gid_list) {
```

Comment:
nit: This could just be `else if` statement in that case ?

Reply:
Yeah, this also looks good to me.